pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java

```diff
@@ -308,13 +309,16 @@ private static boolean inOrder(int[] columnIndices) {
    * @param dataSchema data schema desired for the row.
    * @return canonicalize row.
    */
-  private static Object[] canonicalizeRow(Object[] row, DataSchema dataSchema) {
+  private static Object[] canonicalizeRow(Object[] row, DataSchema dataSchema, DataSchema resultSchema) {
     Object[] resultRow = new Object[row.length];
     for (int colId = 0; colId < row.length; colId++) {
       Object value = row[colId];
       if (value != null) {
         if (dataSchema.getColumnDataType(colId) == DataSchema.ColumnDataType.OBJECT) {
           resultRow[colId] = value;
+        } else if (dataSchema.getColumnDataType(colId) == DataSchema.ColumnDataType.BYTES) {
+          dataSchema.getColumnDataTypes()[colId] = resultSchema.getColumnDataType(colId);
```

Comment:
this seems wrong. we can't simply modify the "desiredDataSchema" which is given during logical planning. modifying it causes downstream issues which might've been hidden due to the auto-casting nature of v2 operator.

Reply:
What should be the approach to handle such cases, where the scalar function return type is `Object` and getting translated to `DataSchema.ColumnDataType.BYTES`?

Reply:
i don't rightfully know. we should add tests against Postgres behavior and see what's expected and modify our behavior accordingly